### PR TITLE
refactor image build process

### PR DESCRIPTION
**Pull request by aaronlevy**
This simplifies the build process a bit (to make it easier to add to CI)

The checkpoint build still requires a 2 step process (build image from commit, then update the desired image in api-server manifest) - but instead uses a template to enforce that the `checkpoint-install` script only ever installs the same version as itself.

/cc @derekparker @pbx0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,14 +22,10 @@ check:
 install: _output/bin/$(LOCAL_OS)/bootkube
 	cp $< $(GOPATH_BIN)
 
-_output/bin/%/bootkube: LDFLAGS=-X github.com/kubernetes-incubator/bootkube/pkg/version.Version=$(shell $(CURDIR)/build/git-version.sh)
-_output/bin/%/bootkube: $(GOFILES)
+_output/bin/%: LDFLAGS=-X github.com/kubernetes-incubator/bootkube/pkg/version.Version=$(shell $(CURDIR)/build/git-version.sh)
```
Comment (colhom) [inline Makefile]: Probably should quote $(CURDIR) in case the absolute path has spaces in it.

File: build/build-image.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+BUILD_IMG=${BUILD_IMG:-}
+
+if [ -z "${BUILD_IMG}" ]; then
+    echo "BUILD_IMG env var must be set"
+    exit 1
+fi
+
+BOOTKUBE_ROOT=$(git rev-parse --show-toplevel)
+source ${BOOTKUBE_ROOT}/build/build-release.sh
```
Comment (colhom) [inline build/build-image.sh]: Also quote any usages of `BOOTKUBE_ROOT` in case it has spaces. Yay bash.

**Comment by derekparker** [comment]
Overall lgtm.

**Comment by pbx0** [comment]
My one nit would be to add some documentation to the scripts about how they work similar to the PR message. Otherwise, LGTM.

**Comment by aaronlevy** [comment]
@pbx0 added some docs

**Comment by pbx0** [comment]
@aaronlevy Nice, docs LGTM.

**Pull request merged**